lib/msf/core/optional_session/smb.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Msf
+  module OptionalSession
+    module SMB
+      include Msf::OptionalSession
+
+      def initialize(info = {})
```

Comment:
I think these protocol-specific `#initialize` methods could be used to add the session's name to the `SessionsType` key of the module info.

That would be something along the lines of:
```
        'SessionTypes'   => %w[SMB],
```

For this one.